diff -urN eccube-2.11.0-orig/data/config/webmatrix.php eccube-2.11.0-IIS-patched/data/config/webmatrix.php
--- ec-cube/data/config/webmatrix.php	1970-01-01 09:00:00.000000000 +0900
+++ ec-cube/data/config/webmatrix.php	2011-03-24 19:34:45.000000000 +0900
@@ -0,0 +1,4 @@
+<?php
+/* WebMatrix Connection String */
+/* mysql://PlaceHolderForUser:PlaceHolderForPassword@PlaceHolderForServer/PlaceHolderForDb;*/
+?>
diff -urN eccube-2.11.0-orig/html/.user.ini eccube-2.11.0-IIS-patched/html/.user.ini
--- ec-cube/html/.user.ini	1970-01-01 09:00:00.000000000 +0900
+++ ec-cube/html/.user.ini	2011-03-24 19:38:06.000000000 +0900
@@ -0,0 +1,9 @@
+mbstring.language = Japanese
+mbstring.encoding_translation = off
+output_handler = NULL
+magic_quotes_gpc = off
+session.auto_start = 0
+mbstring.internal_encoding = UTF-8
+upload_max_filesize = 5M
+register_globals = off
+date.timezone = Asia/Tokyo
--- ec-cube/data/class_extends/util_extends/SC_Utils_Ex.php
+++ ec-cube/data/class_extends/util_extends/SC_Utils_Ex.php
@@ -34,4 +34,21 @@
  */
 class SC_Utils_Ex extends SC_Utils
 {
+    function sfInitInstall() {
+        // インストール済みが定義されていない。
+        if (!defined('ECCUBE_INSTALL')) {
+            $phpself = $_SERVER['SCRIPT_NAME'];
+            if (strpos('/install/', $phpself) === false) {
+                $path = substr($phpself, 0, strpos($phpself, basename($phpself)));
+                $install_url = SC_Utils_Ex::searchInstallerPath($path);
+                header('Location: ' . $install_url);
+                exit;
+            }
+        }
+        $path = HTML_REALDIR . 'install/' . DIR_INDEX_FILE;
+        if (file_exists($path)) {
+            SC_Utils_Ex::sfErrorHeader('&gt;&gt; /install/' . DIR_INDEX_FILE . ' は、インストール完了後にファイルを削除してください。削除するには<a href="' . ROOT_URLPATH . 'deleteInstaller.php">こちら</a>をクリックしてください。');
+        }
+    }
+
 }
--- ec-cube/html/install/index.php
+++ ec-cube/html/install/index.php
@@ -22,6 +22,7 @@
  */
 // ▼require.php 相当
 // rtrim は PHP バージョン依存対策
+define('HTML_REALDIR', rtrim(realpath(rtrim(realpath(dirname(__FILE__)), '/\\') . '/../'), '/\\') . '/');
 $GLOBALS['_realdir'] = rtrim(realpath(rtrim(realpath(dirname(__FILE__)), '/\\') . '/../'), '/\\') . '/';
 $GLOBALS['_realdir'] = str_replace('\\', '/', $GLOBALS['_realdir']);
 $GLOBALS['_realdir'] = str_replace('//', '/', $GLOBALS['_realdir']);
@@ -48,7 +49,6 @@
 
 $objPage = new StdClass;
 $objPage->arrDB_TYPE = array(
-    'pgsql' => 'PostgreSQL',
     'mysql' => 'MySQL',
 );
 $objPage->arrDB_PORT = array(
@@ -684,7 +684,7 @@
     if (defined('MAIL_BACKEND')) {
         $mail_backend = MAIL_BACKEND;
     } else {
-        $mail_backend = 'mail';
+        $mail_backend = 'smtp';
     }
     if (defined('SMTP_HOST')) {
         $smtp_host = SMTP_HOST;
@@ -721,10 +721,47 @@
 // DBパラメーター情報の初期化
 function lfInitDBParam($objDBParam)
 {
+    // EC-CUBEデフォルト値設定
+    $default_server = '127.0.0.1';
+    $default_dbname = 'eccube_db';
+    $default_user   = 'eccube_db_user';
+    $default_pass   = '';
+
+    // WebMatrixの設定ファイルから接続情報を取得
+    $webpi_filename = HTML_REALDIR . HTML2DATA_DIR . 'config/webmatrix.php';
+    if(file_exists($webpi_filename) && $fp = @fopen($webpi_filename, 'r')) {
+        while (!feof($fp)) {
+            $connect_str = fgets($fp);
+            if(preg_match('/mysql/', $connect_str)) {
+                break;
+            }
+        }
+        
+        // MySQLの文字列から分割して接続情報を取得する
+        if(!empty($connect_str)) {
+            // /* mysql://[ユーザー名]:[パスワード]@[ホスト名]/[データベース名];*/
+            // @で分解
+            $split_connect = explode('@', $connect_str);
+
+            // ユーザー名, パスワードを取得
+            $split_userpass_wk = explode('//', $split_connect[0]);
+            $split_userpass    = explode(':', $split_userpass_wk[1]);
+            $default_user      = $split_userpass[0];
+            $default_pass      = $split_userpass[1];
+            
+            // ホスト名, データベース名を取得
+            $split_serverdb_wk = explode(';', $split_connect[1]);
+            $split_serverdb    = explode('/', $split_serverdb_wk[0]);
+            $default_server    = $split_serverdb[0];
+            $default_dbname    = $split_serverdb[1];
+        }
+        fclose($fp);
+    }
+
     if (defined('DB_SERVER')) {
         $db_server = DB_SERVER;
     } else {
-        $db_server = '127.0.0.1';
+        $db_server = $default_server;
     }
 
     if (defined('DB_TYPE')) {
@@ -742,13 +779,19 @@
     if (defined('DB_NAME')) {
         $db_name = DB_NAME;
     } else {
-        $db_name = 'eccube_db';
+        $db_name = $default_dbname;
     }
 
     if (defined('DB_USER')) {
         $db_user = DB_USER;
     } else {
-        $db_user = 'eccube_db_user';
+        $db_user = $default_user;
+    }
+
+    if (defined('DB_PASSWORD')) {
+        $db_password = DB_PASSWORD;
+    } else {
+        $db_password = $default_pass;
     }
 
     $objDBParam->addParam('DBの種類', 'db_type', INT_LEN, '', array('EXIST_CHECK', 'MAX_LENGTH_CHECK'), $db_type);
@@ -756,7 +799,7 @@
     $objDBParam->addParam('DBポート', 'db_port', INT_LEN, '', array('MAX_LENGTH_CHECK'), $db_port);
     $objDBParam->addParam('DB名', 'db_name', MTEXT_LEN, '', array('EXIST_CHECK', 'MAX_LENGTH_CHECK'), $db_name);
     $objDBParam->addParam('DBユーザ', 'db_user', MTEXT_LEN, '', array('EXIST_CHECK', 'MAX_LENGTH_CHECK'), $db_user);
-    $objDBParam->addParam('DBパスワード', 'db_password', MTEXT_LEN, '', array('EXIST_CHECK', 'MAX_LENGTH_CHECK'));
+    $objDBParam->addParam('DBパスワード', 'db_password', MTEXT_LEN, '', array('EXIST_CHECK', 'MAX_LENGTH_CHECK'), $default_pass);
 
     return $objDBParam;
 }
@@ -1026,17 +1069,6 @@
 
     // FIXME 変数出力はエスケープすべき
     $config_data = "<?php\n"
-                 . "define('ECCUBE_INSTALL', 'ON');\n"
-                 . "define('HTTP_URL', '"              . $normal_url . "');\n"
-                 . "define('HTTPS_URL', '"             . $secure_url . "');\n"
-                 . "define('ROOT_URLPATH', '"          . $url_dir . "');\n"
-                 . "define('DOMAIN_NAME', '"           . $objWebParam->getValue('domain') . "');\n"
-                 . "define('DB_TYPE', '"               . $objDBParam->getValue('db_type') . "');\n"
-                 . "define('DB_USER', '"               . $objDBParam->getValue('db_user') . "');\n"
-                 . "define('DB_PASSWORD', '"           . $objDBParam->getValue('db_password') . "');\n"
-                 . "define('DB_SERVER', '"             . $objDBParam->getValue('db_server') . "');\n"
-                 . "define('DB_NAME', '"               . $objDBParam->getValue('db_name') . "');\n"
-                 . "define('DB_PORT', '"               . $objDBParam->getValue('db_port') . "');\n"
                  . "define('ADMIN_DIR', '"             . $objWebParam->getValue('admin_dir') . "/');\n"
                  . "define('ADMIN_FORCE_SSL', "        . $force_ssl . ");\n"
                  . "define('ADMIN_ALLOW_HOSTS', '"     . serialize($allow_hosts) . "');\n"
@@ -1048,7 +1080,7 @@
                  . "define('SMTP_USER', '"             . $objWebParam->getValue('smtp_user') . "');\n"
                  . "define('SMTP_PASSWORD', '"         . $objWebParam->getValue('smtp_password') . "');\n";
 
-    if ($fp = fopen(CONFIG_REALFILE, 'w')) {
+    if ($fp = fopen(DEFINE_REALFILE, 'w')) {
         fwrite($fp, $config_data);
         fclose($fp);
     }
--- ec-cube/data/class/pages/admin/basis/LC_Page_Admin_Basis_ZipInstall.php
+++ ec-cube/data/class/pages/admin/basis/LC_Page_Admin_Basis_ZipInstall.php
@@ -161,10 +161,7 @@
         $objQuery =& SC_Query_Ex::getSingletonInstance();
 
         // DB更新
-        $objQuery->begin();
-        $this->lfDeleteZip();
         $this->insertMtbZip();
-        $objQuery->commit();
     }
 
     /**
@@ -232,34 +229,43 @@
         $cntInsert = 0;
         $img_cnt = 0;
 
-        $fp = $this->openZipCsv();
-        while (!feof($fp)) {
-            $arrCSV = fgetcsv($fp, ZIP_CSV_LINE_MAX);
-            if (empty($arrCSV)) continue;
-            $cntCurrentLine++;
-            if ($cntCurrentLine >= $start) {
-                $sqlval = array();
-                $sqlval['zip_id'] = $cntCurrentLine;
-                $sqlval['zipcode'] = $arrCSV[2];
-                $sqlval['state'] = $arrCSV[6];
-                $sqlval['city'] = $arrCSV[7];
-                $sqlval['town'] = $arrCSV[8];
-                $objQuery->insert('mtb_zip', $sqlval);
+        /* 標準実装が非常に遅いので Azure/WebMatrix 向けにチューニング */
+        $begin = microtime(true);
+
+        try {
+            $dbh = new PDO('mysql:host=' . DB_SERVER . ';dbname=' . DB_NAME . ';port=' . DB_PORT . ';charset=utf8', DB_USER, DB_PASSWORD);
+            $dbh->beginTransaction();
+            $dbh->exec('DELETE FROM mtb_zip');
+
+            $sql = 'INSERT INTO mtb_zip (zip_id, zipcode, state, city, town) VALUES (?, ?, ?, ?, ?)';
+            $stmt = $dbh->prepare($sql);
+            $line = file(ZIP_CSV_UTF8_REALFILE);
+            for($i = 0; $line[$i] != ''; $i ++){
+                if (!($array = explode(",", $line[$i]))) {
+                    continue;
+                }
+                $stmt->bindParam(1, ++$cntCurrentLine, PDO::PARAM_INT);
+                $stmt->bindParam(2, $array[2], PDO::PARAM_STR, strlen($array[2]));
+                $stmt->bindParam(3, $array[6], PDO::PARAM_STR, strlen($array[6]));
+                $stmt->bindParam(4, $array[7], PDO::PARAM_STR, strlen($array[7]));
+                $stmt->bindParam(5, $array[8], PDO::PARAM_STR, strlen($array[8]));
+                $stmt->execute();
                 $cntInsert++;
-            }
 
-            // $disp_line件ごとに進捗表示する
-            if ($cntCurrentLine % $disp_line == 0 && $img_cnt < IMAGE_MAX) {
-                echo '<img src="' . $img_path . 'graph_1_w.gif">';
-                SC_Utils_Ex::sfFlush();
-                $img_cnt++;
+                // $disp_line件ごとに進捗表示する
+                if ($i % $disp_line == 0 && $img_cnt < IMAGE_MAX) {
+                    echo '<img src="' . $img_path . 'graph_1_w.gif">';
+                    SC_Utils_Ex::sfFlush();
+                    $img_cnt++;
+                }
+                SC_Utils_Ex::extendTimeOut();
             }
-            SC_Utils_Ex::extendTimeOut();
-        }
-        fclose($fp);
-
-        echo '<img src="' . $img_path . 'space_w.gif">';
+            $end = microtime(true);
 
+            echo '<img src="' . $img_path . 'space_w.gif">';
+        } catch (PDOException $e) {
+            trigger_error($e->getMessage(), E_USER_ERROR);
+        }
         ?>
         </div>
         <script type='text/javascript' language='javascript'>
@@ -270,7 +276,7 @@
                     document.open('text/html','replace');
                     document.clear();
                     document.write('<p>完了しました。<br />');
-                    document.write("<?php echo $cntInsert ?> 件を追加しました。</p>");
+                    document.write("<?php echo $cntInsert ?> 件を追加しました。<?php echo ($end-$begin); ?> 秒かかりました</p>");
                     document.write("<p><a href='?' target='_top'>戻る</a></p>");
                     document.close();
                 }
@@ -281,6 +287,7 @@
         </body>
         </html>
         <?php
+        $dbh->commit();
     }
 
     public function openZipCsv()
--- ec-cube/data/class/util/GC_Utils.php
+++ ec-cube/data/class/util/GC_Utils.php
@@ -189,6 +189,10 @@
             $path = GC_Utils_Ex::isAdminFunction() ? ADMIN_LOG_REALFILE : LOG_REALFILE;
         }
 
+        if (isset($_SERVER['COMPUTERNAME'])) {
+            $path = str_replace('.log', '.' . $_SERVER['COMPUTERNAME'] . '.log', $path);
+        }
+
         $msg = "$today [{$_SERVER['SCRIPT_NAME']}] $msg from {$_SERVER['REMOTE_ADDR']}\n";
         if ($verbose) {
             if (GC_Utils_Ex::isFrontFunction()) {

--- ec-cube/data/class/SC_AdminView.php
+++ ec-cube/data/class/SC_AdminView.php
@@ -32,8 +32,8 @@
     {
         parent::init();
 
-        $this->_smarty->template_dir = TEMPLATE_ADMIN_REALDIR;
-        $this->_smarty->compile_dir = COMPILE_ADMIN_REALDIR;
+        $this->_smarty->template_dir = realpath(TEMPLATE_ADMIN_REALDIR);
+        $this->_smarty->compile_dir = realpath(COMPILE_ADMIN_REALDIR);
         $this->assign('TPL_URLPATH_PC', ROOT_URLPATH . USER_DIR . USER_PACKAGE_DIR . TEMPLATE_NAME . '/');
         $this->assign('TPL_URLPATH_DEFAULT', ROOT_URLPATH . USER_DIR . USER_PACKAGE_DIR . DEFAULT_TEMPLATE_NAME . '/');
         $this->assign('TPL_URLPATH', ROOT_URLPATH . USER_DIR . USER_PACKAGE_DIR . 'admin/');
	Modified   /var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//KV8stt_SC_InstallView.php
diff --git a/data/class/SC_InstallView.php b//var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//KV8stt_SC_InstallView.php
index c02ee4fee48ea674e4c20d7edf8916dbcf7e645e 100644
--- ec-cube/data/class/SC_InstallView.php
+++ ec-cube/data/class/SC_InstallView.php
@@ -27,7 +27,7 @@
     {
         parent::__construct();
 
-        $this->_smarty->template_dir = $template_dir;
-        $this->_smarty->compile_dir = $compile_dir;
+        $this->_smarty->template_dir = realpath($template_dir);
+        $this->_smarty->compile_dir = realpath($compile_dir);
     }
 }
	Modified   /var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//l6alcC_SC_MobileView.php
diff --git a/data/class/SC_MobileView.php b//var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//l6alcC_SC_MobileView.php
index 8e69e52fd0ef3909c68278636c34953b58cf0d00 100644
--- ec-cube/data/class/SC_MobileView.php
+++ ec-cube/data/class/SC_MobileView.php
@@ -32,8 +32,8 @@
     {
         parent::init();
 
-        $this->_smarty->template_dir = MOBILE_TEMPLATE_REALDIR;
-        $this->_smarty->compile_dir = MOBILE_COMPILE_REALDIR;
+        $this->_smarty->template_dir = realpath(MOBILE_TEMPLATE_REALDIR);
+        $this->_smarty->compile_dir = realpath(MOBILE_COMPILE_REALDIR);
         $this->assignTemplatePath(DEVICE_TYPE_MOBILE);
     }
 }
	Modified   /var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//UUS6tj_SC_SiteView.php
diff --git a/data/class/SC_SiteView.php b//var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//UUS6tj_SC_SiteView.php
index e732a9ec1fd68d24e2ed899162af0ce1cb72c183 100644
--- ec-cube/data/class/SC_SiteView.php
+++ ec-cube/data/class/SC_SiteView.php
@@ -36,8 +36,8 @@
     {
         parent::init();
 
-        $this->_smarty->template_dir = TEMPLATE_REALDIR;
-        $this->_smarty->compile_dir = COMPILE_REALDIR;
+        $this->_smarty->template_dir = realpath(TEMPLATE_REALDIR);
+        $this->_smarty->compile_dir = realpath(COMPILE_REALDIR);
 
         $this->assignTemplatePath(DEVICE_TYPE_PC);
     }
	Modified   /var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//IKWGLp_SC_SmartphoneView.php
diff --git a/data/class/SC_SmartphoneView.php b//var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//IKWGLp_SC_SmartphoneView.php
index 538d4ff87186c2b5dd5dea0f939ecbe195027241 100644
--- ec-cube/data/class/SC_SmartphoneView.php
+++ ec-cube/data/class/SC_SmartphoneView.php
@@ -32,8 +32,8 @@
     {
         parent::init();
 
-        $this->_smarty->template_dir = SMARTPHONE_TEMPLATE_REALDIR;
-        $this->_smarty->compile_dir = SMARTPHONE_COMPILE_REALDIR;
+        $this->_smarty->template_dir = realpath(SMARTPHONE_TEMPLATE_REALDIR);
+        $this->_smarty->compile_dir = realpath(SMARTPHONE_COMPILE_REALDIR);
         $this->assignTemplatePath(DEVICE_TYPE_SMARTPHONE);
     }
 }
diff --git a/html/admin/require.php b//var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//8MD5e1_require.php
index 395f93254af63889f303d6a4aef4096e93c8ca79 100644
--- ec-cube/html/admin/require.php
+++ ec-cube/html/admin/require.php
@@ -22,7 +22,10 @@
  */
 
 // rtrim は PHP バージョン依存対策
-define('HTML_REALDIR', rtrim(realpath(rtrim(realpath(dirname(__FILE__)), '/\\') . '/../'), '/\\') . '/');
+$GLOBALS['_realdir'] = rtrim(realpath(rtrim(realpath(dirname(__FILE__)), '/\\') . '/../'), '/\\') . '/';
+$GLOBALS['_realdir'] = str_replace('\\', '/', $GLOBALS['_realdir']);
+$GLOBALS['_realdir'] = str_replace('//', '/', $GLOBALS['_realdir']);
+define('HTML_REALDIR', $GLOBALS['_realdir']);
 define('ADMIN_FUNCTION', true);
 
 require_once HTML_REALDIR . 'define.php';
	Modified   /var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//H8vceJ_require.php
diff --git a/html/require.php b//var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//H8vceJ_require.php
index 72d99c69dca002c89ab01ad8a26ec085387388c5 100644
--- ec-cube/html/require.php
+++ ec-cube/html/require.php
@@ -22,7 +22,10 @@
  */
 
 // rtrim は PHP バージョン依存対策
-define('HTML_REALDIR', rtrim(realpath(rtrim(realpath(dirname(__FILE__)), '/\\') . '/'), '/\\') . '/');
+$GLOBALS['_realdir'] = rtrim(realpath(rtrim(realpath(dirname(__FILE__)), '/\\') . '/'), '/\\') . '/';
+$GLOBALS['_realdir'] = str_replace('\\', '/', $GLOBALS['_realdir']);
+$GLOBALS['_realdir'] = str_replace('//', '/', $GLOBALS['_realdir']);
+define('HTML_REALDIR', $GLOBALS['_realdir']);
 
 if (!defined('ADMIN_FUNCTION') || ADMIN_FUNCTION !== true) {
     define('FRONT_FUNCTION', true);
--- ec-cube/html/define.php
+++ ec-cube/html/define.php
@@ -1,6 +1,6 @@
 <?php
 /** HTMLディレクトリからのDATAディレクトリの相対パス */
-define('HTML2DATA_DIR', '../data/');
+define('HTML2DATA_DIR', './data/');
 
 /** data/module 以下の PEAR ライブラリを優先的に使用する */
 set_include_path(realpath(dirname(__FILE__) . '/' . HTML2DATA_DIR . 'module') . PATH_SEPARATOR . get_include_path());

--- ec-cube/data/class/SC_Initial.php
+++ ec-cube/data/class/SC_Initial.php
@@ -69,6 +69,7 @@
     public function requireInitialConfig()
     {
         define('CONFIG_REALFILE', realpath(dirname(__FILE__)) . '/../config/config.php');
+        define('DEFINE_REALFILE', realpath(dirname(__FILE__)) . '/../config/define.php');
         if (file_exists(CONFIG_REALFILE)) {
             require_once CONFIG_REALFILE;
         }
--- ec-cube/data/class/db/dbfactory/SC_DB_DBFactory_MYSQL.php
+++ ec-cube/data/class/db/dbfactory/SC_DB_DBFactory_MYSQL.php
@@ -362,5 +362,6 @@
     {
         $objQuery->exec('SET SESSION storage_engine = InnoDB');
         $objQuery->exec("SET SESSION sql_mode = 'ANSI'");
+        $objQuery->exec("SET SESSION time_zone = '+09:00'"); // Asia/Tokyo
     }
 }
--- ec-cube/data/config/config.php
+++ ec-cube/data/config/config.php
@@ -0,0 +1,38 @@
+<?php
+$realpath = dirname(__FILE__);
+define('ROOT_URLPATH', '/');
+
+// for Azure
+if (strpos($_SERVER['SERVER_NAME'], 'localhost') === false) {
+    $location = "//" . $_SERVER["SERVER_NAME"] . ROOT_URLPATH;
+    $http_location = 'http:' . $location;
+    $https_location = 'https:' . $location;
+}
+// for WebMatrix
+else {
+    $http_location = "http://" . $_SERVER["SERVER_NAME"] . ":" . $_SERVER["SERVER_PORT"] . ROOT_URLPATH;
+    $https_location = $http_location;
+}
+
+$define_php = $realpath . './define.php';
+$webmatrix_php = $realpath . './webmatrix.php';
+if (file_exists($define_php)) {
+    require_once($define_php);
+
+    if (file_exists($webmatrix_php)) {
+        $subject = file_get_contents($webmatrix_php);
+        preg_match("|/\\*\\s*mysql://([^:]*):([^@]*)@([^/]*)/([^;]*);\\*/|", $subject, $matches);
+        list($all, $db_user, $db_password, $db_server, $db_name) = $matches;
+
+        define('ECCUBE_INSTALL', 'ON');
+        define('HTTP_URL', $http_location);
+        define('HTTPS_URL', $https_location);
+        define('DOMAIN_NAME', '');
+        define('DB_TYPE', 'mysql');
+        define('DB_USER', $db_user);
+        define('DB_PASSWORD', $db_password);
+        define('DB_SERVER', $db_server);
+        define('DB_NAME', $db_name);
+        define('DB_PORT', '');
+    }
+}
--- ec-cube/html/deleteInstaller.php
+++ ec-cube/html/deleteInstaller.php
@@ -0,0 +1,16 @@
+<?php
+
+require_once '../data/config/config.php';
+if (defined('ECCUBE_INSTALL')) {
+    $installerFile = './install/index.php';
+    if (file_exists($installerFile)) {
+        unlink($installerFile);
+    }
+}
+$url = 'http://' . urlencode($_SERVER['SERVER_NAME']) . '/';
+if (isset($_SERVER['HTTP_REFERER'])
+    && preg_match('!^(https?://[^\?]+)!', $_SERVER['HTTP_REFERER'], $m)
+) {
+    $url = $m[1];
+}
+header("Location: {$url}");
--- ec-cube/html/install/templates/step1.tpl
+++ ec-cube/html/install/templates/step1.tpl
@@ -21,7 +21,7 @@
  *}-->
 <script type="text/javascript">
 $(function() {
-    $('.option').hide();
+    // $('.option').hide();
     if ($('input[name=mail_backend]').val() == 'smtp') {
         $('.smtp').attr('disabled', false);
     } else {
@@ -144,7 +144,7 @@
                     <td>
                     <!--{assign var=key value="normal_url"}-->
                     <span class="attention"><!--{$arrErr[$key]}--></span>
-                    <input type="text" name="<!--{$key}-->" value="<!--{$arrForm[$key].value|h}-->" maxlength="<!--{$arrForm[$key].length}-->" style="<!--{$arrErr[$key]|sfGetErrorColor}-->" size="50"  />
+                    <input type="text" name="<!--{$key}-->" value="<!--{$arrForm[$key].value|h}-->" maxlength="<!--{$arrForm[$key].length}-->" style="<!--{$arrErr[$key]|sfGetErrorColor}-->" size="50"  readonly="readonly" />
                     </td>
                 </tr>
                 <tr>
@@ -152,7 +152,7 @@
                     <td>
                     <!--{assign var=key value="secure_url"}-->
                     <span class="attention"><!--{$arrErr[$key]}--></span>
-                    <input type="text" name="<!--{$key}-->" value="<!--{$arrForm[$key].value|h}-->" maxlength="<!--{$arrForm[$key].length}-->" style="<!--{$arrErr[$key]|sfGetErrorColor}-->" size="50"  />
+                    <input type="text" name="<!--{$key}-->" value="<!--{$arrForm[$key].value|h}-->" maxlength="<!--{$arrForm[$key].length}-->" style="<!--{$arrErr[$key]|sfGetErrorColor}-->" size="50" readonly="readonly" />
                     </td>
                 </tr>
                 <tr>
@@ -168,7 +168,7 @@
 
             <p><a href="javascript:;" id="options">&gt;&gt; オプション設定</a></p>
             <div class="option">
-                <h2>メールサーバーの設定(オプション)</h2>
+                <h2>メールサーバーの設定</h2>
                 <table>
                     <col width="30%" />
                     <col width="70%" />
