diff -urN eccube-2.11.0-orig/data/config/webmatrix.php eccube-2.11.0-IIS-patched/data/config/webmatrix.php
--- ec-cube/data/config/webmatrix.php	1970-01-01 09:00:00.000000000 +0900
+++ ec-cube/data/config/webmatrix.php	2011-03-24 19:34:45.000000000 +0900
@@ -0,0 +1,4 @@
+<?php
+/* WebMatrix Connection String */
+/* mysql://PlaceHolderForUser:PlaceHolderForPassword@PlaceHolderForServer/PlaceHolderForDb;*/
+?>
diff -urN eccube-2.11.0-orig/html/.user.ini eccube-2.11.0-IIS-patched/html/.user.ini
--- ec-cube/html/.user.ini	1970-01-01 09:00:00.000000000 +0900
+++ ec-cube/html/.user.ini	2011-03-24 19:38:06.000000000 +0900
@@ -0,0 +1,9 @@
+mbstring.language = Japanese
+mbstring.encoding_translation = off
+output_handler = NULL
+magic_quotes_gpc = off
+session.auto_start = 0
+mbstring.internal_encoding = UTF-8
+upload_max_filesize = 5M
+register_globals = off
+date.timezone = Asia/Tokyo
--- ec-cube/data/class_extends/util_extends/SC_Utils_Ex.php
+++ ec-cube/data/class_extends/util_extends/SC_Utils_Ex.php
@@ -34,4 +34,21 @@
  */
 class SC_Utils_Ex extends SC_Utils
 {
+    function sfInitInstall() {
+        // インストール済みが定義されていない。
+        if (!defined('ECCUBE_INSTALL')) {
+            $phpself = $_SERVER['SCRIPT_NAME'];
+            if (strpos('/install/', $phpself) === false) {
+                $path = substr($phpself, 0, strpos($phpself, basename($phpself)));
+                $install_url = SC_Utils_Ex::searchInstallerPath($path);
+                header('Location: ' . $install_url);
+                exit;
+            }
+        }
+        $path = HTML_REALDIR . 'install/' . DIR_INDEX_FILE;
+        if (file_exists($path)) {
+            SC_Utils_Ex::sfErrorHeader('&gt;&gt; /install/' . DIR_INDEX_FILE . ' は、インストール完了後にファイルを削除してください。削除するには<a href="' . ROOT_URLPATH . 'deleteInstaller.php">こちら</a>をクリックしてください。');
+        }
+    }
+
 }
--- ec-cube/html/install/index.php
+++ ec-cube/html/install/index.php
@@ -45,7 +45,6 @@
 
 $objPage = new StdClass;
 $objPage->arrDB_TYPE = array(
-    'pgsql' => 'PostgreSQL',
     'mysql' => 'MySQL',
 );
 $objPage->arrDB_PORT = array(
@@ -718,10 +717,47 @@
 // DBパラメーター情報の初期化
 function lfInitDBParam($objDBParam)
 {
+    // EC-CUBEデフォルト値設定
+    $default_server = '127.0.0.1';
+    $default_dbname = 'eccube_db';
+    $default_user   = 'eccube_db_user';
+    $default_pass   = '';
+
+    // WebMatrixの設定ファイルから接続情報を取得
+    $webpi_filename = HTML_REALDIR . HTML2DATA_DIR . 'config/webmatrix.php';
+    if(file_exists($webpi_filename) && $fp = @fopen($webpi_filename, 'r')) {
+        while (!feof($fp)) {
+            $connect_str = fgets($fp);
+            if(preg_match('/mysql/', $connect_str)) {
+                break;
+            }
+        }
+        
+        // MySQLの文字列から分割して接続情報を取得する
+        if(!empty($connect_str)) {
+            // /* mysql://[ユーザー名]:[パスワード]@[ホスト名]/[データベース名];*/
+            // @で分解
+            $split_connect = explode('@', $connect_str);
+
+            // ユーザー名, パスワードを取得
+            $split_userpass_wk = explode('//', $split_connect[0]);
+            $split_userpass    = explode(':', $split_userpass_wk[1]);
+            $default_user      = $split_userpass[0];
+            $default_pass      = $split_userpass[1];
+            
+            // ホスト名, データベース名を取得
+            $split_serverdb_wk = explode(';', $split_connect[1]);
+            $split_serverdb    = explode('/', $split_serverdb_wk[0]);
+            $default_server    = $split_serverdb[0];
+            $default_dbname    = $split_serverdb[1];
+        }
+        fclose($fp);
+    }
+
     if (defined('DB_SERVER')) {
         $db_server = DB_SERVER;
     } else {
-        $db_server = '127.0.0.1';
+        $db_server = $default_server;
     }
 
     if (defined('DB_TYPE')) {
@@ -739,13 +775,19 @@
     if (defined('DB_NAME')) {
         $db_name = DB_NAME;
     } else {
-        $db_name = 'eccube_db';
+        $db_name = $default_dbname;
     }
 
     if (defined('DB_USER')) {
         $db_user = DB_USER;
     } else {
-        $db_user = 'eccube_db_user';
+        $db_user = $default_user;
+    }
+
+    if (defined('DB_PASSWORD')) {
+        $db_password = DB_PASSWORD;
+    } else {
+        $db_password = $default_pass;
     }
 
     $objDBParam->addParam('DBの種類', 'db_type', INT_LEN, '', array('EXIST_CHECK', 'MAX_LENGTH_CHECK'), $db_type);
@@ -753,7 +795,7 @@
     $objDBParam->addParam('DBポート', 'db_port', INT_LEN, '', array('MAX_LENGTH_CHECK'), $db_port);
     $objDBParam->addParam('DB名', 'db_name', MTEXT_LEN, '', array('EXIST_CHECK', 'MAX_LENGTH_CHECK'), $db_name);
     $objDBParam->addParam('DBユーザ', 'db_user', MTEXT_LEN, '', array('EXIST_CHECK', 'MAX_LENGTH_CHECK'), $db_user);
-    $objDBParam->addParam('DBパスワード', 'db_password', MTEXT_LEN, '', array('EXIST_CHECK', 'MAX_LENGTH_CHECK'));
+    $objDBParam->addParam('DBパスワード', 'db_password', MTEXT_LEN, '', array('EXIST_CHECK', 'MAX_LENGTH_CHECK'), $default_pass);
 
     return $objDBParam;
 }
--- ec-cube/data/class/pages/admin/basis/LC_Page_Admin_Basis_ZipInstall.php	(revision 23350)
+++ ec-cube/data/class/pages/admin/basis/LC_Page_Admin_Basis_ZipInstall.php	(working copy)
@@ -232,31 +232,33 @@
         $cntInsert = 0;
         $img_cnt = 0;
 
-        $fp = $this->openZipCsv();
-        while (!feof($fp)) {
-            $arrCSV = fgetcsv($fp, ZIP_CSV_LINE_MAX);
-            if (empty($arrCSV)) continue;
-            $cntCurrentLine++;
-            if ($cntCurrentLine >= $start) {
-                $sqlval = array();
-                $sqlval['zip_id'] = $cntCurrentLine;
-                $sqlval['zipcode'] = $arrCSV[2];
-                $sqlval['state'] = $arrCSV[6];
-                $sqlval['city'] = $arrCSV[7];
-                $sqlval['town'] = $arrCSV[8];
-                $objQuery->insert('mtb_zip', $sqlval);
-                $cntInsert++;
+        $begin = microtime(true);
+        $con = $objQuery->conn->connection;
+        $stmt = 'INSERT INTO mtb_zip (zip_id, zipcode, state, city, town) VALUES (?, ?, ?, ?, ?)';
+        mysql_query('PREPARE insert_mtb_zip FROM \'' . mysql_real_escape_string($stmt) . '\'', $con);
+        $line = file(ZIP_CSV_UTF8_REALFILE);
+        for($i = 0; $line[$i] != ''; $i ++){
+            if (!($array = explode(",", $line[$i]))) {
+                continue;
             }
+            mysql_query('SET @zip_id = ' . mysql_real_escape_string(++$cntCurrentLine, $con), $con);
+            mysql_query('SET @zipcode = \'' . mysql_real_escape_string($array[2], $con) . '\'', $con);
+            mysql_query('SET @state = \'' . mysql_real_escape_string($array[6], $con) . '\'', $con);
+            mysql_query('SET @city = \'' . mysql_real_escape_string($array[7], $con) . '\'', $con);
+            mysql_query('SET @town = \'' . mysql_real_escape_string($array[8], $con) . '\'', $con);
+            mysql_query('EXECUTE insert_mtb_zip USING @zip_id, @zipcode, @state, @city, @town', $con);
+            $cntInsert++;
 
             // $disp_line件ごとに進捗表示する
-            if ($cntCurrentLine % $disp_line == 0 && $img_cnt < IMAGE_MAX) {
+            if ($i % $disp_line == 0 && $img_cnt < IMAGE_MAX) {
                 echo '<img src="' . $img_path . 'graph_1_w.gif">';
                 SC_Utils_Ex::sfFlush();
                 $img_cnt++;
             }
             SC_Utils_Ex::extendTimeOut();
         }
-        fclose($fp);
+        mysql_query('DEALLOCATE PREPARE insert_mtb_zip');
+        $end = microtime(true);
 
         echo '<img src="' . $img_path . 'space_w.gif">';
 
@@ -270,7 +272,7 @@
                     document.open('text/html','replace');
                     document.clear();
                     document.write('<p>完了しました。<br />');
-                    document.write("<?php echo $cntInsert ?> 件を追加しました。</p>");
+                    document.write("<?php echo $cntInsert ?> 件を追加しました。<?php echo ($end-$begin); ?> 秒かかりました</p>");
                     document.write("<p><a href='?' target='_top'>戻る</a></p>");
                     document.close();
                 }
--- ec-cube/data/class/util/GC_Utils.php
+++ ec-cube/data/class/util/GC_Utils.php
@@ -189,6 +189,10 @@
             $path = GC_Utils_Ex::isAdminFunction() ? ADMIN_LOG_REALFILE : LOG_REALFILE;
         }
 
+        if (isset($_SERVER['COMPUTERNAME'])) {
+            $path = str_replace('.log', '.' . $_SERVER['COMPUTERNAME'] . '.log', $path);
+        }
+
         $msg = "$today [{$_SERVER['SCRIPT_NAME']}] $msg from {$_SERVER['REMOTE_ADDR']}\n";
         if ($verbose) {
             if (GC_Utils_Ex::isFrontFunction()) {

--- ec-cube/data/class/SC_AdminView.php
+++ ec-cube/data/class/SC_AdminView.php
@@ -32,8 +32,8 @@
     {
         parent::init();
 
-        $this->_smarty->template_dir = TEMPLATE_ADMIN_REALDIR;
-        $this->_smarty->compile_dir = COMPILE_ADMIN_REALDIR;
+        $this->_smarty->template_dir = realpath(TEMPLATE_ADMIN_REALDIR);
+        $this->_smarty->compile_dir = realpath(COMPILE_ADMIN_REALDIR);
         $this->assign('TPL_URLPATH_PC', ROOT_URLPATH . USER_DIR . USER_PACKAGE_DIR . TEMPLATE_NAME . '/');
         $this->assign('TPL_URLPATH_DEFAULT', ROOT_URLPATH . USER_DIR . USER_PACKAGE_DIR . DEFAULT_TEMPLATE_NAME . '/');
         $this->assign('TPL_URLPATH', ROOT_URLPATH . USER_DIR . USER_PACKAGE_DIR . 'admin/');
	Modified   /var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//KV8stt_SC_InstallView.php
diff --git a/data/class/SC_InstallView.php b//var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//KV8stt_SC_InstallView.php
index c02ee4fee48ea674e4c20d7edf8916dbcf7e645e 100644
--- ec-cube/data/class/SC_InstallView.php
+++ ec-cube/data/class/SC_InstallView.php
@@ -27,7 +27,7 @@
     {
         parent::__construct();
 
-        $this->_smarty->template_dir = $template_dir;
-        $this->_smarty->compile_dir = $compile_dir;
+        $this->_smarty->template_dir = realpath($template_dir);
+        $this->_smarty->compile_dir = realpath($compile_dir);
     }
 }
	Modified   /var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//l6alcC_SC_MobileView.php
diff --git a/data/class/SC_MobileView.php b//var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//l6alcC_SC_MobileView.php
index 8e69e52fd0ef3909c68278636c34953b58cf0d00 100644
--- ec-cube/data/class/SC_MobileView.php
+++ ec-cube/data/class/SC_MobileView.php
@@ -32,8 +32,8 @@
     {
         parent::init();
 
-        $this->_smarty->template_dir = MOBILE_TEMPLATE_REALDIR;
-        $this->_smarty->compile_dir = MOBILE_COMPILE_REALDIR;
+        $this->_smarty->template_dir = realpath(MOBILE_TEMPLATE_REALDIR);
+        $this->_smarty->compile_dir = realpath(MOBILE_COMPILE_REALDIR);
         $this->assignTemplatePath(DEVICE_TYPE_MOBILE);
     }
 }
	Modified   /var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//UUS6tj_SC_SiteView.php
diff --git a/data/class/SC_SiteView.php b//var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//UUS6tj_SC_SiteView.php
index e732a9ec1fd68d24e2ed899162af0ce1cb72c183 100644
--- ec-cube/data/class/SC_SiteView.php
+++ ec-cube/data/class/SC_SiteView.php
@@ -36,8 +36,8 @@
     {
         parent::init();
 
-        $this->_smarty->template_dir = TEMPLATE_REALDIR;
-        $this->_smarty->compile_dir = COMPILE_REALDIR;
+        $this->_smarty->template_dir = realpath(TEMPLATE_REALDIR);
+        $this->_smarty->compile_dir = realpath(COMPILE_REALDIR);
 
         $this->assignTemplatePath(DEVICE_TYPE_PC);
     }
	Modified   /var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//IKWGLp_SC_SmartphoneView.php
diff --git a/data/class/SC_SmartphoneView.php b//var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//IKWGLp_SC_SmartphoneView.php
index 538d4ff87186c2b5dd5dea0f939ecbe195027241 100644
--- ec-cube/data/class/SC_SmartphoneView.php
+++ ec-cube/data/class/SC_SmartphoneView.php
@@ -32,8 +32,8 @@
     {
         parent::init();
 
-        $this->_smarty->template_dir = SMARTPHONE_TEMPLATE_REALDIR;
-        $this->_smarty->compile_dir = SMARTPHONE_COMPILE_REALDIR;
+        $this->_smarty->template_dir = realpath(SMARTPHONE_TEMPLATE_REALDIR);
+        $this->_smarty->compile_dir = realpath(SMARTPHONE_COMPILE_REALDIR);
         $this->assignTemplatePath(DEVICE_TYPE_SMARTPHONE);
     }
 }
diff --git a/html/admin/require.php b//var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//8MD5e1_require.php
index 395f93254af63889f303d6a4aef4096e93c8ca79 100644
--- ec-cube/html/admin/require.php
+++ ec-cube/html/admin/require.php
@@ -22,7 +22,10 @@
  */
 
 // rtrim は PHP バージョン依存対策
-define('HTML_REALDIR', rtrim(realpath(rtrim(realpath(dirname(__FILE__)), '/\\') . '/../'), '/\\') . '/');
+$GLOBALS['_realdir'] = rtrim(realpath(rtrim(realpath(dirname(__FILE__)), '/\\') . '/../'), '/\\') . '/';
+$GLOBALS['_realdir'] = str_replace('\\', '/', $GLOBALS['_realdir']);
+$GLOBALS['_realdir'] = str_replace('//', '/', $GLOBALS['_realdir']);
+define('HTML_REALDIR', $GLOBALS['_realdir']);
 define('ADMIN_FUNCTION', true);
 
 require_once HTML_REALDIR . 'define.php';
	Modified   /var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//83X5Au_index.php
diff --git a/html/install/index.php b//var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//83X5Au_index.php
index a39c2c53d9dd25ccd608523116403ec9735ccc78 100644
--- ec-cube/html/install/index.php
+++ ec-cube/html/install/index.php
@@ -23,6 +23,10 @@
 // ▼require.php 相当
 // rtrim は PHP バージョン依存対策
 define('HTML_REALDIR', rtrim(realpath(rtrim(realpath(dirname(__FILE__)), '/\\') . '/../'), '/\\') . '/');
+$GLOBALS['_realdir'] = rtrim(realpath(rtrim(realpath(dirname(__FILE__)), '/\\') . '/../'), '/\\') . '/';
+$GLOBALS['_realdir'] = str_replace('\\', '/', $GLOBALS['_realdir']);
+$GLOBALS['_realdir'] = str_replace('//', '/', $GLOBALS['_realdir']);
+define('HTML_REALDIR', $GLOBALS['_realdir']);
 
 require_once HTML_REALDIR . 'define.php';
 define('INSTALL_FUNCTION', true);
	Modified   /var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//H8vceJ_require.php
diff --git a/html/require.php b//var/folders/0w/tmwgmlh968n5q05dbkr5hf4c0000gn/T//H8vceJ_require.php
index 72d99c69dca002c89ab01ad8a26ec085387388c5 100644
--- ec-cube/html/require.php
+++ ec-cube/html/require.php
@@ -22,7 +22,10 @@
  */
 
 // rtrim は PHP バージョン依存対策
-define('HTML_REALDIR', rtrim(realpath(rtrim(realpath(dirname(__FILE__)), '/\\') . '/'), '/\\') . '/');
+$GLOBALS['_realdir'] = rtrim(realpath(rtrim(realpath(dirname(__FILE__)), '/\\') . '/'), '/\\') . '/';
+$GLOBALS['_realdir'] = str_replace('\\', '/', $GLOBALS['_realdir']);
+$GLOBALS['_realdir'] = str_replace('//', '/', $GLOBALS['_realdir']);
+define('HTML_REALDIR', $GLOBALS['_realdir']);
 
 if (!defined('ADMIN_FUNCTION') || ADMIN_FUNCTION !== true) {
     define('FRONT_FUNCTION', true);

